### ISSUE TITLE
HiDPI scaling works only on even resolutions

### ISSUE THREAD

**Issue by kultigspritzig** [syscl/XPS9350-macOS#52]
The HiDPI scaling works only on half resolution. This applies to my external 4k display as well. Any other resolution will have a standard, 'ugly' scaling applied.

Is this a known issue?

**Comment by syscl** [comment]
What do you mean ```Half resolution``` ?

Retina will scale your resolution, that's by design. For example, if you use QHD(3200x1800), then macOS will scale 3200x1800 to 1600x900(HiDPI). On real MacBook, there's some properties being injected by system, thus we will see a cosmetic 2880x1400 in ```System Information```->```Graphics``` which is actually 1440x720(HiDPI). 

If you compare QHD vs. MacBook's Retina, you will soon know which is better. Apple failed in this case.

That's it,
syscl

**Comment by kultigspritzig** [comment]
I don't think we understand each other. 

What I mean is that scaling does not use all the available pixels for any resolution that is not full or half of the original. Real MacBook will do that. I can clearly see a difference in clarity of the image between my laptop and the Mac next to it. 

System Information -> Graphics shows fixed 1600 x 900. This doesn't change. The only good looking option in Display settings is "Looks like 1600 x 900" - this one has proper HiDPI applied. Any other option there does not have HiDPI - it scales to 1600 x 900. I can clearly see how the details of wallpaper disappear. They should *not*.

**Comment by syscl** [comment]
try this [one](https://github.com/syscl/Enable-HiDPI-OSX) . Don't forget to backup your ```/System/Library/Displays/Contents/Resources/Overrides/``` before executing ```./enable-HiDPI.sh```. Once the system corrupt, you have a way to rescue from Recovery HD

syscl

**Comment by kultigspritzig** [comment]
Thanks, I will definitely give it a try.

**Comment by thiscodedbox** [comment]
I like this utility for more granular control over HiDPI settings:

Retina Display Manager

https://github.com/avibrazil/RDM

**Comment by kultigspritzig** [comment]
@thiscodedbox Thanks, that helped for the built-in display. Now I can have an equivalent of 1920x1080 with HiDPI enabled, which I could not achieve with regular Display panel.

It doesn't do anything for my external display, though - still only 1920x1080 is the highest that provides HiDPI on a 3840 x 2160 display.

**Comment by kultigspritzig** [comment]
Actually, I just noticed that - I get HiDPI enabled at 1920x1080 only if... I have my external display connected. Otherwise there's no HiDPI. This is messed up.

**Comment by thiscodedbox** [comment]
Weird! Did you try the retina utility I linked?

sent from mobile device. Please excuse any brevity.

On Feb 25, 2017 11:44, "kultigspritzig" <notifications@github.com> wrote:

Actually, I just noticed that - I get HiDPI enabled at 1920x1080 only if...
I have my external display connected. Otherwise there's no HiDPI. This is
messed up.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/syscl/XPS9350-macOS/issues/52#issuecomment-282507415>,
or mute the thread

**Comment by kultigspritzig** [comment]
@thiscodedbox yes, that's the one I use. I have no time to spare to investigate this I am afraid.

**Comment by flipphos** [comment]
Retina Display Manager works on my XPS 9350, 3200x1800 screen. Thanks for the tip @thiscodedbox
